examples/07_end_to_end/earthquake.ipynb

```diff
@@ -0,0 +1,416 @@
+    "                        )\n",
+    "                        # For a PushPullAdapter, push_tick takes 3 arguments: push_tick(live: bool, time: datetime, value: csp.Struct)\n",
+    "                        self.push_tick(live, my_event.time.datetime, my_event)\n",
+    "                interval = self._interval.total_seconds()\n",
+    "                time.sleep(interval)\n",
+    "            else:\n",
+    "                for event in catalog:\n",
+    "                    time.sleep(0.2)\n",
+    "                    if event not in history:\n",
+    "                        history.append(event)\n",
+    "                        my_event = EventData(\n",
+    "                            time=event.origins[0].time,\n",
+    "                            longitude=event.origins[0].longitude,\n",
+    "                            latitude=event.origins[0].latitude,\n",
+    "                            magnitude=event.magnitudes[0].mag,\n",
+    "                            creation_time=event.origins[0].creation_info.creation_time.datetime,\n",
+    "                        )\n",
+    "                        # For a PushPullAdapter, push_tick takes 3 arguments: push_tick(live: bool, time: datetime, value: csp.Struct)\n",
+    "                        self.push_tick(live, now, my_event)\n",
+    "                self.stop()\n",
+    "\n",
+    "# Create the graph-time representation of our adapter\n",
+    "FetchEventData = py_pushpull_adapter_def(\"FetchEventData\", FetchEventDataAdapter, csp.ts[EventData], interval=timedelta, url=str, realtime=bool)\n",
+    "\n",
+    "@csp.node\n",
+    "def update_widget(event: csp.ts[EventData], widget: widgets.widgets.widget_selectioncontainer.Tab):\n",
+    "    if csp.ticked(event):\n",
+    "        # widget.children = [datagrid, worldmap]\n",
+    "        data = {\n",
+    "            \"time\": [event.time],\n",
+    "            \"longitude\": [event.longitude],\n",
+    "            \"latitude\": [event.latitude],\n",
+    "            \"magnitude\": [event.magnitude],\n",
+    "            \"creation_time\": [event.creation_time],\n",
+    "        }\n",
+    "        widget.children[0].update(data)\n",
+    "        widget.children[1].update(data)\n",
+    "\n",
+    "@csp.graph\n",
+    "def earthquake_graph():\n",
+    "    print(\"Start of graph building\")\n",
+    "    url = \"https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_day.quakeml\"\n",
+    "    interval = timedelta(seconds=60)\n",
+    "    earthquakes = FetchEventData(interval, url=url, realtime=csp.is_configured_realtime())\n",
+    "    update_widget(earthquakes, widget=widget)\n",
+    "    csp.add_graph_output(\"Earthquakes\", earthquakes)\n",
+    "    print(\"End of graph building\")\n",
+    "\n",
+    "start = datetime.utcnow()\n",
+    "end = start + timedelta(minutes=10)\n",
+    "csp.run(earthquake_graph, starttime=start, endtime=end, realtime=False)\n",
```

Comment:
given what I said above, the example should start some time in the past ( say a few hours ) and end some time in the future ( say timedelta(minutes=30 ) with realtime=True

Reply:
Got it, will rework